nsxt/data_source_nsxt_policy_transit_gateway.go

```diff
@@ -18,14 +20,39 @@ func dataSourceNsxtPolicyTransitGateway() *schema.Resource {
 			"description":  getDataSourceDescriptionSchema(),
 			"path":         getPathSchema(),
 			"context":      getContextSchemaExtended(true, false, false, true),
+			"is_default": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: false,
+			},
 		},
 	}
 }
 
 func dataSourceNsxtPolicyTransitGatewayRead(d *schema.ResourceData, m interface{}) error {
-	_, err := policyDataSourceResourceRead(d, getPolicyConnector(m), getSessionContext(d, m), "TransitGateway", nil)
+	// Using deprecated API because GetOk is not behaving as expected when is_default = "false".
+	// It does not return true for a key that's explicitly set to false.
+	_, defaultOK := d.GetOkExists("is_default")
+	_, dpOk := d.GetOk("display_name")
+	_, idOk := d.GetOk("id")
+
+	if defaultOK && !dpOk && !idOk {
```

Comment:
This behavior should be documented, that is_default and the other attributes are mutually exclusive.